monkey/monkey_island/cc/services/authentication_service.py

```diff
@@ -21,6 +28,14 @@ def __init__(
         self._repository_encryptor = repository_encryptor
         self._island_event_queue = island_event_queue
 
+    def needs_registration(self) -> bool:
+        """
+        Checks if a user is already registered on the Island
+
+        :return: Whether registration is required on the Island
+        """
+        return not User.objects.first()
+
     def register_new_user(self, username: str, password: str):
```

Comment:
flask-security already has registration built in. If you set `app.config["SECURITY_REGISTER_URL"] = "/api/register"` you might need to do minimal changes to the UI. The problem is that username login is supported, but not registration. I think the best way is to override `ConfirmRegisterForm` or `UniqueEmailFormMixin` to not require the email for validation. In that case we would get all the benefits of validation and normalization.

Reply:
I'll look into it.